src/springfield/fields.py

```diff
@@ -390,12 +390,26 @@ def __init__(self, entity, *args, **kwargs):
                        Use 'self' to use the :class:`Entity` class that
                        this field is already bound to.
         """
-        self.type = entity
+        self._type = entity
         super(EntityField, self).__init__(*args, **kwargs)
 
+    @property
+    def type(self):
+        if not callable(self._type):
+            try:
+                if '.' in self._type:
+                    modules, _kls_name = self._type.rsplit('.', 1)
+                    _module = __import__(modules, fromlist=[_kls_name])
+                    _kls = getattr(_module, _kls_name) 
+                    assert callable(_kls), 'Dotted-name entity types should be callable.'
+                    self._type = _kls
+            except Exception as e:
+                raise ValueError('Invalid class name for EntityField: %s (%s)' % (self._type, e))
```

Comment:
The original traceback is lost which could make debugging this import error problematic. Keep the traceback:

```
try:
    buggy_code_which_throws_exception()
except Exception as e:
    raise Exception, "The code is buggy: %s" % e, sys.exc_info()[2]
```